remote-config/src/test/kotlin/com/rakuten/tech/mobile/remoteconfig/ConfigCacheSpec.kt

```diff
@@ -125,11 +126,45 @@ class ConfigCacheSpec : RobolectricBaseSpec() {
 
         When calling stubVerifier.verify(any()) itReturns false
 
-        val cache = ConfigCache(stubFetcher, createFile(filename), stubPoller, stubVerifier)
+        val cache = ConfigCache(stubFetcher, createFile(filename), stubPoller, stubVerifier, false)
 
         cache["foo"] shouldEqual null
     }
 
+    @Test
+    fun `should not apply after fetching`() {
+        val filename = "cache.json"
+
+        When calling stubFetcher.fetch() itReturns createConfig(hashMapOf("foo" to "bar"))
+        When calling stubVerifier.verify(any()) itReturns true
+        val cache = ConfigCache(stubFetcher, createFile(filename), AsyncPoller(1), stubVerifier, false)
```

Comment:
You should probably use a mock of the `AsyncPoller` here instead of relying on the real implementation. See the `create cache with fetched config` test below. Then yo can set the mock poller to delay before invoking the function.

Reply:
Used mock `AsycPoller`